kafka-rest/src/main/java/io/confluent/kafkarest/resources/v2/PartitionsResource.java

```diff
@@ -200,13 +216,124 @@ public void onCompletion(
     );
   }
 
-  private boolean topicExists(final String topic)  throws Exception {
-    return ctx.getAdminClientWrapper().topicExists(topic);
+  /**
+   * Get offsets for the given {@code topic} and {@code partition}.
+   *
+   * <p>If {@code timestamp} is not specified, returns a summary with beginning and end offsets.
+   * Otherwise, returns the earliest offset whose timestamp is after the given {@code timestamp} and
+   * its associated timestamp, or empty if no such offset exists.</p>
+   *
+   * @throws io.confluent.rest.exceptions.RestNotFoundException if either {@code topic} or {@code
+   *                                                            partition} don't exist.
+   */
+  @GET
+  @Path("/{partition}/offsets")
+  public GetTopicPartitionOffsetResponse getOffsets(
+      @PathParam("topic") String topic,
+      @PathParam("partition") int partition,
+      @QueryParam("timestamp") @Nullable Instant timestamp
+  ) throws Exception {
+    checkTopicExists(topic);
```

Comment:
these will both result in synchronous calls to the cluster which seems undesirable - i think you want to just rely on errors returned by the offset requests.

Reply:
I believe the Java API will block until the topic/partition exists. I think that's why the other actions do it as well. The option is to have a timeout. But since this action is not going to resource intensive, having the check before might make it fail faster then relying on the timeout. Re-open if you don't agree.

Reply:
ahh right, this is following what other resources do, fair enough.

not good, but also not something to try and address in this PR - belongs in a separate PR that fixes it everywhere.